packages/devp2p/src/eth/index.ts

```diff
@@ -230,7 +231,15 @@ export class ETH extends EventEmitter {
         this._peer._socket.remotePort
       } (eth${this._version}): ${this._getStatusString(this._status)}`
     )
-    this._send(ETH.MESSAGE_CODES.STATUS, rlp.encode(this._status as any))
+
+    let payload = rlp.encode(this._status as any)
+
+    // Use snappy compression if peer supports DevP2P >=v5
+    if ((this._peer._hello?.protocolVersion ?? 0) >= 5) {
+      payload = snappy.compress(payload)
+    }
+
+    this._send(ETH.MESSAGE_CODES.STATUS, payload)
```

Comment:
I think fundamentally you are correct in that Snappy is an RLPx protocol level change and not within subprotocols.  The challenge I had when initially coding this was that when I added the snappy compression check in `Peer._sendMessage`, I couldn't find a clear way to identify when the message being sent was part of the handshake (i.e. `HELLO/DISCONNECT/PING/PONG`) versus a subprotocol specific message since the `code` passed to `_sendMessage` can indicate either an RLPx handshake message or any of the first 4 messages sent by a subprotocol and `_sendMessage` can be called directly from the subprotocol handler.  That said, maybe I'm missing something here and this is more straight forward than I'm thinking.

Reply:
hmm, that is an interesting problem! one idea could be to add a param to  Peer._sendMessage() `fromSubprotocol = false` to get better granularity on `code`.

Reply:
Ok, I wasn't aware of the differentiation here. So I guess it's also ok to keep as an alternative to add these kind of new flags or something, this is not bloating the structure or anything and a very local addition.